skbeam/core/speckle.py

```diff
@@ -237,20 +240,29 @@ def _process(num_roi, level, buf_no, buf, img_per_level, labels,
         probability density of detecting photons
     prob_k_pow : array
         squares of probability density of detecting photons
+    track_bad : array
+        to track bad images in each level
     """
     img_per_level[level] += 1
     u_labels = list(np.unique(labels))
 
-    for j, label in enumerate(u_labels):
-        roi_data = buf[level, buf_no][labels == label]
-        spe_hist, bin_edges = np.histogram(roi_data, bins=bin_edges,
-                                           density=True)
-        spe_hist = np.nan_to_num(spe_hist)
-        prob_k[level, j] += (spe_hist -
-                             prob_k[level, j])/(img_per_level[level])
-
-        prob_k_pow[level, j] += (np.power(spe_hist, 2) -
-                                 prob_k_pow[level, j])/(img_per_level[level])
+    # check if there are any bad images, using bad_to_nan function
```

Comment:
bad_to_nan function is a separate function in mask.py. If there is bad images, (and defining bad is up to the user), user can use bad_to_nan function for images that will be used in XSVS or XPCS analysis. Therefore, if they used it for the images, this XSVS function has to check and remove that image and track it, if the user didn't use it(all the images are good) then the old xsvs analysis. No change